src/_includes/layout.njk

```diff
@@ -7,11 +7,118 @@
+         To prevent layout shifts that can result from this render, we use a
+         "dsd-pending" attribute to ensure we only paint after we know
+         shadow DOM is active. -->
+    <style>
+      body[dsd-pending] {
+        display: none;
+      }
+    </style>
   </head>
-  <body>
+  <body dsd-pending>
+    <script>
+      if (HTMLTemplateElement.prototype.hasOwnProperty('shadowRoot')) {
+        // This browser has native declarative shadow DOM support, so we can
+        // allow painting immediately.
+        document.body.removeAttribute('dsd-pending');
+      }
+    </script>
+
     {% include "header.njk" %}
     {{ content | safe }}
+
+    <!-- At this point, browsers with native shadow DOM support will already
+         be able to paint the initial fully styled state your components,
+         without executing a single line of JavaScript! However, the components
+         aren't interactive yet - that's what hydration is for. -->
+
+    <!-- Use a type=module script so that we can use dynamic module imports.
+         Note this pattern will not work in IE11. -->
+    <script type="module-shim">
+      (async () => {
+        // Start fetching the Lit hydration support module (note the absence
+        // of "await" -- we don't want to block yet).
+        const litHydrateSupportInstalled = import(
+          './node_modules/lit-element/experimental-hydrate-support.js'
+        );
+
+        // Check if we require the declarative shadow DOM polyfill. As of
+        // February 2022, Chrome and Edge have native support, but Firefox
+        // and Safari don't yet.
+        if (!HTMLTemplateElement.prototype.hasOwnProperty('shadowRoot')) {
+          // Fetch the declarative shadow DOM polyfill.
+          const {hydrateShadowRoots} = await import(
+            './node_modules/@webcomponents/template-shadowroot/template-shadowroot.js'
+          );
+
+          // Apply the polyfill. This is a one-shot operation, so it is important
+          // it happens after all HTML has been parsed.
+          hydrateShadowRoots(document.body);
+
+          // At this point, browsers without native declarative shadow DOM
+          // support can paint the initial state of your components!
+          document.body.removeAttribute('dsd-pending');
+        }
+
+        // The Lit hydration support module must be installed before we can
+        // load any component definitions. Wait until it's ready.
+        await litHydrateSupportInstalled;
+
+        // Load component definitions. As each component definition loads, your
+        // pre-rendered components will come to life and become interactive.
+        //
+        // You may also prefer to bundle your components into fewer JS modules.
+        // See https://lit.dev/docs/tools/production/#building-with-rollup for
+        // more details.
+        {% for path in wc.filePaths %}
+        import('{{ path }}');
+        {% endfor %}
+
+        {% if archiveConfig.runtimeOnlyArchives %}
```

Comment:
This is where `runtimeOnlyArchives` gets loaded; the rest of the changes in this file is mostly pulled from https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit